data/database.py

```diff
@@ -2035,6 +2038,12 @@ class RepositoryAutoPrunePolicy(BaseModel):
     policy = JSONField(null=False, default={})
 
 
+class TagNotificationSuccess(BaseModel):
+    notification = ForeignKeyField(RepositoryNotification, index=True, null=False)
+    tag = ForeignKeyField(Tag, index=True, null=False)
+    method = ForeignKeyField(ExternalNotificationMethod, null=False)
```

Comment:
You'll have to test when ever the foreign notification or tag rows are deleted that this row get's deleted as well without causing any issues.

Reply:
yes, I added both in the second PR, [tags clean up](https://github.com/quay/quay/pull/2895/files#diff-d54089ecfb2c9ed6312fb03ca1ea1cd1be9555c8dedf9138a2feac65ea530ff5R496) and [notification clean up](https://github.com/quay/quay/pull/2895/files#diff-8f5c31fa9c15f136ba9dd9d240d5c8b265556d6a2246f7af9f30df806543f3caR246)